### PR TITLE
chore: Add upgrade guide [skip ci]

### PR DESCRIPTION
**Pull request by royjit**
*Issue #, if available:*

*Description of changes:*

*Check points: (check or cross out if not relevant)*

- [ ] Added new tests to cover change, if needed
- [ ] Build succeeds with all target using Swift Package Manager
- [ ] All unit tests pass
- [ ] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [ ] PR title conforms to conventional commit style
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: UpgradeGuide.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+As part of the latest version of Amplify Library for Swift, we have re-written our APIs to support idiomatic Swift features. 
+
+### Structured Concurrency (async/await pattern)
+
+All our APIs have been refactored to follow the async/await structured concurrency pattern. The following is an example of how you would implement Sign-In for your authentication flows:
+
+v1 - Callback api
+
+```
```
Comment (ruisebas) [inline UpgradeGuide.md:11]: Maybe add `swift` to all code blocks so we get syntax highlight? 

e.g. having `swift`:
```swift
func signIn(username: String, password: String) {
    Amplify.Auth.signIn(username: username, password: password) { result in
        switch result {
        case .success:
            print("Sign in succeeded")
        case .failure(let error):
            print("Sign in failed \(error)")
        }
    }
}
```

vs without:

```
func signIn(username: String, password: String) {
    Amplify.Auth.signIn(username: username, password: password) { result in
        switch result {
        case .success:
            print("Sign in succeeded")
        case .failure(let error):
            print("Sign in failed \(error)")
        }
    }
}
```

File: UpgradeGuide.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
+# Upgrade Guide: Amplify Library for Swift
+
+As part of the latest version of Amplify Library for Swift, we have re-written our APIs to support idiomatic Swift features. 
+
+### Structured Concurrency (async/await pattern)
+
+All our APIs have been refactored to follow the async/await structured concurrency pattern. The following is an example of how you would implement Sign-In for your authentication flows:
+
+v1 - Callback api
```
Comment (hdworld11) [inline UpgradeGuide.md:9]:
```suggestion
v1 - Callback API
```

**Review by ruisebas — CHANGES_REQUESTED** [review]

File: UpgradeGuide.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    }
+}
+```
+
+### Combine support
+
+Support for combine apis was also changed in the v2. You can read more about the changes in the dedicated doc here - 
+https://github.com/aws-amplify/amplify-swift/blob/main/README-combine-support.md
+
+For detailed definitions of all the Amplify features available in this version, please refer to o[ur latest v2 documentation](https://docs.amplify.aws/lib/q/platform/ios/).
```
Comment (hdworld11) [inline UpgradeGuide.md:43]:
```suggestion
For detailed definitions of all the Amplify features available in this version, please refer to [our latest v2 documentation](https://docs.amplify.aws/lib/q/platform/ios/).
```

File: UpgradeGuide.md (hunk 1 above)
Comment (royjit) [reply UpgradeGuide.md]: updated

**Review by hdworld11 — APPROVED** [review]

**Review by ruisebas — APPROVED** [review]

**Pull request merged**